Source/ASDisplayNode+Layout.mm

```diff
@@ -352,8 +352,10 @@ - (void)_locked_measureNodeWithBoundsIfNecessary:(CGRect)bounds
     ASLayout *layout = [self calculateLayoutThatFits:constrainedSize
                                     restrictedToSize:self.style.size
                                 relativeToParentSize:boundsSizeForLayout];
-    
     nextLayout = std::make_shared<ASDisplayNodeLayout>(layout, constrainedSize, boundsSizeForLayout, version);
+    // Now that the constrained size of pending layout might have been reused, the layout is useless
+    // Release it and any orphaned subnodes it retains
+    _pendingDisplayNodeLayout = nullptr;
```

Comment:
Either of this line or the new layout version check [above](https://github.com/TextureGroup/Texture/pull/455/files#diff-71ea742a29543cb7e08c3dffe2a2508eR308) fixes part of the problem. The reason I included it is that at this point, the pending layout can be stale ([here](https://github.com/TextureGroup/Texture/pull/455/files#diff-71ea742a29543cb7e08c3dffe2a2508eR338)) and it's never going to be applied. And yet, it still holds onto orphaned subnodes that were either never inserted or removed by newer layouts (check out `-[ASLayout retainSublayoutLayoutElements]`).